### PR TITLE
Linode post/put/delete docs cleanup

### PR DESCRIPTION
**Pull request by SzySteve**
#### What:
Cleanup linode endpoint docs by adding missing example parameters and documenting rules about resizing disks

### CODE REVIEW

File: docs/src/data/endpoints/linodes.yaml [hunk 1]
```diff
@@ -367,10 +370,12 @@ endpoints:
               Sets the name of the image shown in the base image list, defaults
               to the source disk label.
             type: String
+            value: My gold master image
           description:
             optional: true
             description: An optional description of the created image
             type: String
+            value: This is the image description, its more descriptive than a label.
```
Comment (yoshizzle) [inline docs/src/data/endpoints/linodes.yaml]: it's or it is

File: docs/src/data/endpoints/linodes.yaml [hunk 2]
```diff
@@ -916,8 +921,9 @@ endpoints:
         params:
           type:
             description: >
-                An IP Address Type for this IP Address. Public IP's incur a monthly cost.
-            type: string
+                An IP Address Type for this IP Address: either public or private. Public IP's incur a monthly cost.
```
Comment (yoshizzle) [inline docs/src/data/endpoints/linodes.yaml]: Comma after Address me thinks.

Do we also want to markdown "public" or "private" to make it more obvious that those are values that should be passed in?

Not sure if we have a convention for this throughout the docs, but we probably should.

Comment (SzySteve) [reply docs/src/data/endpoints/linodes.yaml]: Unsure if we can do that in this section: I havent seen a way to do it without just verbosely printing my syntax. Ill use quotes for now.

File: docs/src/data/endpoints/linodes.yaml [hunk 3]
```diff
@@ -309,14 +310,15 @@ endpoints:
     group: Disks
     type: Strange
     authenticated: true
-    description: Resizes the disk.
+    description: Resizes the disk. The linode this disk is attached to must be shut down for this to take effect.
```
Comment (RobertDeRose) [inline docs/src/data/endpoints/linodes.yaml]: To take effect or required to work?

File: docs/src/data/endpoints/linodes.yaml [hunk 4]
```diff
@@ -417,7 +422,8 @@ endpoints:
             value: My config
           devices:
             description: >
-              Disks attached to this Linode config.
+              Disks attached to this Linode config. Devices may have either a volume or a disk configured,
```
Comment (RobertDeRose) [inline docs/src/data/endpoints/linodes.yaml]: This is confusing.  I would probably reword this as such

```
Devices that would be attached to a Linode when booted using this config.  Devices can be either Disks or Volumes, but only one type per slot (a - h).
```

File: docs/src/data/endpoints/linodes.yaml [hunk 5]
```diff
@@ -916,8 +921,9 @@ endpoints:
         params:
           type:
             description: >
-                An IP Address Type for this IP Address. Public IP's incur a monthly cost.
-            type: string
+                An IP Address Type for this IP Address, either "public" or "private". Public IP's incur a monthly cost.
```
Comment (RobertDeRose) [inline docs/src/data/endpoints/linodes.yaml]:
```
The type of IP Address this is, can be one of "public" or "private". Public IP Addresses, over and above the one included with each Linode, incur an additional monthly charge.
```

File: docs/src/data/endpoints/linodes.yaml (hunk 4 above)
Comment (RobertDeRose) [inline docs/src/data/endpoints/linodes.yaml]: This is confusing.  I would probably reword this as such

```
Devices that would be attached to a Linode when booted using this config.  Devices can be either Disks or Volumes, but only one type per slot (a - h).
```

File: docs/src/data/endpoints/linodes.yaml (hunk 3 above)
Comment (displague) [reply docs/src/data/endpoints/linodes.yaml]: Linodes are considered proper nouns. Instances are not.

File: docs/src/data/endpoints/linodes.yaml [hunk 6]
```diff
@@ -309,14 +310,15 @@ endpoints:
     group: Disks
     type: Strange
     authenticated: true
-    description: Resizes the disk.
+    description: Resizes the disk. The Linode this disk is attached to must be shut down for this to take effect.
```
Comment (yoshizzle) [inline docs/src/data/endpoints/linodes.yaml]: can you change the second _this_ to _resizing_, please.

File: docs/src/data/endpoints/linodes.yaml [hunk 7]
```diff
@@ -417,7 +422,8 @@ endpoints:
             value: My config
           devices:
             description: >
-              Disks attached to this Linode config.
+              Devices that would be attached to a Linode when booted using this config.
+              Devices can be either Disks or Volumes, but only one type per slot (a - h).
```
Comment (Dorthu) [inline docs/src/data/endpoints/linodes.yaml]: I know you didn't do this, but the example for creating a config show sda and sdb both being POSTed in as volumes - could you make one of them a disk so the example shows how to do both?

screen shot 2017-11-07 at 9 43 47 am

File: docs/src/data/endpoints/linodes.yaml [hunk 8]
```diff
@@ -916,8 +921,10 @@ endpoints:
         params:
           type:
             description: >
-                An IP Address Type for this IP Address. Public IP's incur a monthly cost.
-            type: string
+                The type of IP Address this is, can be one of "public" or "private".
+                Public IP Addresses, over and above the one included with each Linode, incur an additional monthly charge.
```
Comment (Dorthu) [inline docs/src/data/endpoints/linodes.yaml]: might also be worth noting that additional public ip addresses may require you to open a Support Ticket justifying the address before you're allowed to create it (the api returns an error to the tune of "You must request an additional IP Address - please open a ticket")

File: docs/src/data/endpoints/linodes.yaml (hunk 7 above)
Comment (SzySteve) [reply docs/src/data/endpoints/linodes.yaml]: ATM I cannot figure out how to do this: that is defined by the
```
response_type:device
```
which references `objects/device.yaml`, which I have atm configured to only return a volume id not a disk id since it can only do one. Previously the docs showed these as filled in with both disk and volume ID.

Comment (Dorthu) [reply docs/src/data/endpoints/linodes.yaml]: that's fine, thanks for looking
